node/src/tasks/bsp_upload_file.rs

```diff
@@ -517,29 +517,52 @@ where
                     anyhow::anyhow!("Failed to query storage provider capacity: {:?}", e)
                 })?;
 
+            // we registered it to the queue
+            let mut capacity_queue = self.capacity_queue.lock().await;
+
+            capacity_queue.push(event.size);
+
+            drop(capacity_queue);
+
             // Wait for the earliest block where the capacity can be changed.
             self.storage_hub_handler
                 .blockchain
                 .wait_for_block(earliest_change_capacity_block)
                 .await?;
 
-            self.storage_hub_handler
-                .blockchain
-                .send_extrinsic(call, Tip::from(0))
-                .await?
-                .with_timeout(Duration::from_secs(
-                    self.storage_hub_handler
-                        .provider_config
-                        .extrinsic_retry_timeout,
-                ))
-                .watch_for_success(&self.storage_hub_handler.blockchain)
-                .await?;
+            // wwe read from the queue
+            let mut capacity_queue = self.capacity_queue.lock().await;
 
-            info!(
-                target: LOG_TARGET,
-                "Increased storage capacity to {:?} bytes",
-                new_capacity
-            );
+            // if the queue is not empty it is that the capacity hasn't been updated yet
+            if !capacity_queue.is_empty() {
+                let size: u64 = capacity_queue.drain(..).sum();
+
+                let new_capacity = self.calculate_capacity(size, current_capacity)?;
```

Comment:
I couldn't find a good way to test this, especially for the max storage value because it would require a 4GB file and it is not practical.

A way this function could be tested is to pass the config as an argument and have a test in rust to make sure the calculation is right and that it will goes higher than the max storage authorize.

Reply:
You could pass these aditional arguments in your test and reduce it to a reasonable size.

https://github.com/Moonsong-Labs/storage-hub/blob/64cc252115ed75b9d01e52576949f8ef43619051/test/suites/integration/bsp/onboard.test.ts#L22-L25